app/controllers/buyers/applications_controller.rb

```diff
@@ -68,19 +64,20 @@ def new
 
   # TODO: this should be done by buy! method
   def create
+    @plans = accessible_plans
+    application_plan = @plans.find(params[:cinstance][:plan_id])
```

Comment:
I think that we should enforce those params to be present 😄 
```suggestion
    application_plan = @plans.find params.require(:cinstance).permit(:plan_id).tap do |plan_params|
      plan_params.require(:plan_id)
    end
```